### PR TITLE
[DPE-4951] - fix: re-enable prefixed topic names during relations

### PR DESCRIPTION
**Pull request by marcoppenheimer**
#### `fix: re-enable prefixed topic names during relations`
- Related applications will be granted ACL permissions for the requested topic, as a prefix (i.e all topic names starting with th passed field
- The requirements for `topic-name` switch from `LITERAL` ACLs to `PREFIXED` ACLs are:
    - 4 chars or more
    - Ends with a `*` char

### CODE REVIEW

**Review by deusebio — APPROVED** [review]
Looks very good! Thanks!

File: src/managers/auth.py [hunk 1]
```diff
@@ -229,12 +229,20 @@ def add_acl(
         ]
 
         if resource_type == "TOPIC":
-            command += [f"--topic={resource_name}"]
+            if len(resource_name) > 3 and resource_name.endswith("*"):
```
Comment (Batalex) [inline src/managers/auth.py:232]: **question**: Is there a need for more validation using a regex to ensure that the resource is indeed 3 chars and a *, and not some other value?
Something like `^[a-z]{3,}\*$`

File: tests/unit/test_auth.py [hunk 2]
```diff
@@ -112,3 +112,38 @@ def test_add_user_adds_zk_tls_flag(harness: Harness[KafkaCharm]):
             in args["bin_args"]
         ), "--zk-tls-config-file flag not found"
         assert "--zookeeper=" in args["bin_args"], "--zookeeper flag not found"
+
+
+def test_prefixed_acls(harness: Harness[KafkaCharm]):
+    """Checks the requirements for adding and removing PREFIXED ACLs."""
+    with (patch("workload.KafkaWorkload.run_bin_command") as patched_run_bin,):
```
Comment (Batalex) [inline tests/unit/test_auth.py:119]: **nitpick**: Non blocking, but if you revisit this file, maybe we can remove this single element tuple

File: tests/unit/test_auth.py [hunk 3]
```diff
@@ -112,3 +112,38 @@ def test_add_user_adds_zk_tls_flag(harness: Harness[KafkaCharm]):
...
+        for func in [
+            harness.charm.broker.auth_manager.add_acl,
+            harness.charm.broker.auth_manager.remove_acl,
+        ]:
+            func(
+                username="bilbo",
+                operation="READ",
+                resource_type="TOPIC",
+                resource_name="there-and-back-again",
```
Comment (Batalex) [inline tests/unit/test_auth.py:128]: **praise**: I just love our LOTR themed tests so much

**Review by Batalex — APPROVED** [review]
I am ok with merging this as-is, but maybe you'll find my regex suggestion interesting

File: tests/unit/test_auth.py (hunk 3 above)
Comment (marcoppenheimer) [reply tests/unit/test_auth.py:128]: Anything to make writing them less frustrating :P

File: tests/unit/test_auth.py (hunk 2 above)
Comment (marcoppenheimer) [reply tests/unit/test_auth.py:119]: Done.

File: src/managers/auth.py (hunk 1 above)
Comment (deusebio) [reply src/managers/auth.py:232]: I agree with Marc. Given if it is a fairly simple use case, I would also prioritize readability. I would use regex when one has more complex patterns with a nice comment explaining what the regex does. Although powerful, I find regex one of the most exoteric syntaxes

**Pull request merged**